### DISCUSSION TITLE
Is it possible to remove the unresponsive time of sound effects?

### DISCUSSION THREAD

**Discussion by HubKing** [ihhub/fheroes2#6617] [Q&A]
For example, if you build a thing in a castle, the thing appears in a fade-in fashion and some sound effect gets playing until the sound effects ends, the UI does not respond. That is, you can't click the Exit button. The same is true for a lot of other things. I wonder if this behaviour can be changed?

**Comment by oleg-derevenetz** [comment]
Well, that's how these things work in the original game - you can't do anything until the animation (for example, the building construction) finishes.

**Reply by HubKing** [reply]
Yeah, I guess it was like that in the original, but wasn't it because of technical reason of the hardware at that time? Is there any reason why the user should not be allowed to exit the castle before the animation/sound effect ends?

**Reply by zenseii** [reply]
I've checked just now and as @oleg-derevenetz said this is the exact behavior of the original in all of these cases like castle building, puzzle map revealing, spawning after using dimension door or portals etc.

Having said that, many other users have requested this change for all of these situations mentioned, and later Heroes games do allow you to interrupt this - e.g. in Heroes 4 when you build something you can open any of the dialogs in the castle or exit the castle screen and the sound will play until it finishes in the background.

I personally would welcome such a change since most of the time you visit a town it is just to build something and leave and end your turn.
